src/Blubrry/REST/Resource/Social.php

```diff
@@ -49,6 +68,85 @@ public function updateListing($program_keyword, $body) {
+
+        $social_types = ['twitter', 'youtube', 'facebook'];
+        $required_fields = ['social-id', 'social-image', 'social-title'];
+        $attributes = [
+            'html' => [
+                'required' => [],
+                'optional' => ['content'],
+            ],
+            'input-text' => [
+                'required' => ['label', 'name'],
+                'optional' => ['placeholder', 'help-text', 'rows', 'maxlength', 'value'],
+            ],
+            'input-checkbox' => [
+                'required' => [],
+                'optional' => ['label', 'checked', 'name', 'value'],
+            ],
+            'input-radio' => [
+                'required' => [],
+                'optional' => ['label', 'checked', 'name', 'value'],
+            ],
+        ];
+
+        foreach ($social_types as $type) {
+            foreach ($params[$type] as $item) {
+                $SocialOption = [];
+
+                foreach ($required_fields as $field) {
+                    if (!empty($item[$field])) {
+                        $socialOption[$field] = $item[$field];
+                    } else {
+                        return false;
+                    }
+                }
+                
+                $SocialOption['social-type'] = $type;
+                $SocialOption['form-data'] = [];
+
+                if (is_array($item['form-data'])) {
+                    if (empty($item['form-data']['field-type'])) {
+                        return false;
+                    }
+
+                    if (empty($item['form-data']['field-order'])) {
+                        return false;
+                    }
+
+                    $type = $item['form-data']['field-type'];
+                    $order = $item['form-data']['field-order'];
+
+                    foreach ($attributes as $attr) {
+                        if ($attr == $type) {
+                            foreach ($attr['required'] as $req) {
+                                if (empty($item['form-data'][$req])) {
+                                    return false;
+                                } else {
+                                    $SocialOption['form-data'][$req] = $item['form-data'][$req];
+                                }
+                            }
+
+                            foreach ($attr['optional'] as $opt) {
+                                if (empty($item['form-data'][$opt])) {
+                                    continue;
+                                } else {
+                                    $SocialOption['form-data'][$opt] = $item['form-data'][$opt];
+                                }
+                            }
+                        }
+                    }
+
+                    $SocialOption['form-data']['field-type'] = $type;
+                    $SocialOption['form-data']['field-order'] = $order;
+                }
+
+                $social_options['social-options'][] = $SocialOption;
```

Comment:
@lucashillebrandt Sometimes this option is called `socialOption`, sometimes `SocialOptions`, and sometimes `social_option`. I suspect that's incorrect, so can you please confirm?

The PSR2 standard is camel case, so `socialOption` would be the way to go here.

Reply:
Fixed variables names.